user-guide/Advanced_Modules/User Defined APIs/Defining_an_API/UD_APIs_Examples.md

```diff
@@ -0,0 +1,245 @@
+
+            var chain = engine.FindElement(elementName);
+            chain.SetParameter(parameterName, value);
+        }
+    }
+}
+```
+
+## Other examples
+
+### Using the RequestMethod and Route
+
+The `RequestMethod` and `Route` properties on the `ApiTriggerInput` can be used in your script to determine what actions should be executed. This allows you to build a single API script which can be used for multiple APIs. In the example below you can see that the script is used for two similar routes. It also only allows specific request methods depending on the route.
+
+```csharp
+using Skyline.DataMiner.Automation;
+using Skyline.DataMiner.Net.Apps.UserDefinableApis.Actions;
+using Skyline.DataMiner.Net.Apps.UserDefinableApis;
+
+namespace UserDefinableApiScripts.Examples.RequestMethodAndRoute
+{
+    public class Script
+    {
+        [AutomationEntryPoint(AutomationEntryPointType.Types.OnApiTrigger)]
+        public ApiTriggerOutput OnApiTrigger(IEngine engine, ApiTriggerInput requestData)
+        {
+            switch (requestData.Route)
+            {
+                case "chains/change-status":
+                    return HandleStateChangeRequest(requestData);
+
+                case "chains/status":
+                    return HandleStateRetrievalRequests(requestData);
+            }
+
+            return new ApiTriggerOutput
+            {
+                ResponseCode = (int) StatusCode.NotFound
+            };
+        }
+
+        private ApiTriggerOutput HandleStateChangeRequest(ApiTriggerInput requestData)
+        {
+            if (requestData.RequestMethod != RequestMethod.Post)
+            {
+                return new ApiTriggerOutput
+                {
+                    ResponseCode = (int) StatusCode.MethodNotAllowed // We only allow POST
+                };
+            }
+
+            // Handle request...
+        }
+
+        private ApiTriggerOutput HandleStateRetrievalRequests(ApiTriggerInput requestData)
+        {
+            // Handle request...
+        }
+    }
+}
+```
+
+### Starting long running actions asynchronous
+
+When you would want to trigger the execution of a long running action or script, it is recommended to execute this asynchronously and not block the HTTP trigger until the action is completed. The example below shows how you can start an asynchronous subscript in your entry point method. It will thus not be possible to return any result of this subscript since the HTTP call will immediately return after we started it.
+
+If you would want to know whether the action is completed, you could implement a second API that polls a part of your solution which was altered by this long running action to see if the that action was finished. This could for example be the status of a DOM instance, status of a booking or a value of a parameter on a specific element. There is currently no reliable built in way of knowing whether the subscript was finished.
```

Comment:
```suggestion
If you would want to know whether the action is completed, you could implement a second API that polls a part of your solution which was altered by this long running action to see if the action was finished. This could for example be the status of a DOM instance, status of a booking or a value of a parameter on a specific element. There is currently no reliable built in way of knowing whether the subscript was finished.
```